File: OSCAR/src/control/Controller.cpp
#include "Controller.h"

void Controller::configureGame() {

	std::shared_ptr<Player> p1 = std::make_shared<Player>("Pedro[1]", 100);
	_game->addPlayerToTable(p1);
	
	std::shared_ptr<Player> p2 = std::make_shared<Player>("Pedro[2]", 100);
	_game->addPlayerToTable(p2);

	std::shared_ptr<Player> p3 = std::make_shared<Player>("Pedro[3]", 100);
	_game->addPlayerToTable(p3);

	std::shared_ptr<Player> p4 = std::make_shared<Player>("Pedro[4]", 100);
	_game->addPlayerToTable(p4);

	std::shared_ptr<Player> p5 = std::make_shared<Player>("Pedro[5]", 100);
	_game->addPlayerToTable(p5);

	std::shared_ptr<Player> p6 = std::make_shared<Player>("Pedro[6]", 100);
	_game->addPlayerToTable(p6);

	std::cout << "Asignando roles a los jugadores...\n";
	_game->assignRolesToAllPlayers();
}

Controller::Controller(std::unique_ptr<Game> game) {
	_game = std::move(game);
}

void Controller::run() {

	configureGame();
	
	std::cout << "Comienza la partida!\n";
	int i = 0;
	while (i == 0 && !_game->isGameFinished()) {
		_game->playHand();
		_game->passTurn();
		++i;
	}

	std::cout << "Fin de la partida!\n";
}

File: tfgpoker/server/src/main/java/com/ucm/control/Controller.java
package com.ucm.control;

import java.util.List;

import java.io.IOException;

import com.ucm.exceptions.OnlyOnePlayerLeftException;
import com.ucm.gameobjects.Player;
import com.ucm.logic.Game;
import com.ucm.middleclasses.ClientStructGame;


public class Controller {

    /**
     * Atributo que referencia la clase Game
     */
    private Game _game;

    public Controller(Game game, List<ClientStructGame> players) {
        _game = game;
        addPlayersToGame(players);
    }

    private void addPlayersToGame(List<ClientStructGame> players) {

        int id = 0;
        for(ClientStructGame cs : players){
            _game.addPlayer( new Player(id, cs.name(), cs.socket(), 1000) );
            ++id;
        }

        while(true){}
    }

    public void run() {

        // Game loop
        _game.assignRolesToAllPlayers();
        while (!_game.isGameFinished()) {

            try {
                
                // Pre-flop
                _game.shareOutCardsToAllPlayers();
                if (Game.DEBUG)
                    _game.showStateDEBUG();
                _game.playHand();

                // Flop
                _game.addCardToTable();
                _game.addCardToTable();
                _game.addCardToTable();
                if (Game.DEBUG)
                    _game.showStateDEBUG();
                _game.playHand();

                // Turn
                _game.addCardToTable();
                if (Game.DEBUG)
                    _game.showStateDEBUG();
                _game.playHand();

                // River
                _game.addCardToTable();
                if (Game.DEBUG)
                    _game.showStateDEBUG();
                _game.playHand();

                // Showdown
                _game.giveRewardToWinner();
                if (Game.DEBUG)
                    _game.showStateDEBUG();
            }
            catch (OnlyOnePlayerLeftException e) {
                _game.giveRewardToWinner();
            }

            // Devolver todas las cartas al mazo, restablecer jugadores que han 'foldeado',
            // etc...
            _game.restartRound();
            _game.passTurn();
        }
    }

}
